fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -250,20 +250,12 @@ def copy_logfile(device, log_identity, logs_destination_dir)
       end
 
       def copy_logarchive(device, log_identity, logs_destination_dir)
-        sim_resource_dir = FastlaneCore::CommandExecutor.execute(command: "xcrun simctl getenv #{device.udid} SIMULATOR_SHARED_RESOURCES_DIRECTORY 2>/dev/null", print_all: false, print_command: true)
-        logarchive_src = File.join(sim_resource_dir, "system_logs.logarchive")
-        logarchive_dst = File.join(logs_destination_dir, "system_logs-#{log_identity}.logarchive")
+        require 'shellwords'
 
-        # if logarchive already exists it fails as the .logarchive is a directory, so delete it. to be sure its gone
-        FileUtils.rm_rf(logarchive_src)
+        logarchive_dst = Shellwords.escape(File.join(logs_destination_dir, "system_logs-#{log_identity}.logarchive"))
         FileUtils.rm_rf(logarchive_dst)
-
-        command = "xcrun simctl spawn #{device.udid} log collect 2>/dev/null"
+        command = "xcrun simctl spawn #{device.udid} log collect --output #{logarchive_dst} 2>/dev/null"
         FastlaneCore::CommandExecutor.execute(command: command, print_all: false, print_command: true)
-
-        FileUtils.mkdir_p(logarchive_dst)
-        FileUtils.cp_r("#{logarchive_src}/.", logarchive_dst)
-        UI.success "Copying file '#{logarchive_src}' to '#{logarchive_dst}'..."
```

Comment:
I had thought of using the return value from the `execute`, but looking at this, we are already printing the command, so I don't think this is necessary.